packages/builders/src/node/execute/node-execute.builder.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  BuildEvent,
+  Builder,
+  BuilderConfiguration,
+  BuilderContext
+} from '@angular-devkit/architect';
+import { ChildProcess, fork } from 'child_process';
+import * as treeKill from 'tree-kill';
+
+import { Observable, bindCallback, of } from 'rxjs';
+import { concatMap, tap, map } from 'rxjs/operators';
+
+import {
+  BuildNodeBuilderOptions,
+  NodeBuildEvent
+} from '../build/node-build.builder';
+
+export interface NodeExecuteBuilderOptions {
+  inspect: boolean;
+  args: string[];
+  buildTarget: string;
+}
+
+export class NodeExecuteBuilder implements Builder<NodeExecuteBuilderOptions> {
+  private subProcess: ChildProcess;
+
+  constructor(private context: BuilderContext) {}
+
+  run(
+    target: BuilderConfiguration<NodeExecuteBuilderOptions>
+  ): Observable<BuildEvent> {
+    const options = target.options;
+
+    return this.startBuild(options).pipe(
+      concatMap((event: NodeBuildEvent) => {
+        if (event.success) {
+          return this.restartProcess(event.outfile, options).pipe(
+            map(() => {
+              return event;
+            })
+          );
+        } else {
+          this.context.logger.error(
+            'There was an error with the build. See above.'
+          );
+          this.context.logger.info(`${event.outfile} was not restarted.`);
+          return of(event);
+        }
+      })
+    );
+  }
+
+  private runProcess(file: string, options: NodeExecuteBuilderOptions) {
+    if (this.subProcess) {
+      throw new Error('Already running');
+    }
+    this.subProcess = fork(file, options.args, {
+      execArgv: options.inspect ? ['--inspect'] : []
+    });
+  }
+
+  private restartProcess(file: string, options: NodeExecuteBuilderOptions) {
+    const obs$ = this.subProcess ? this.killProcess() : of(undefined);
+    return obs$.pipe(
+      tap(() => {
+        this.runProcess(file, options);
+      })
+    );
+  }
+
+  private killProcess() {
+    const observableTreeKill = bindCallback(treeKill);
+    return observableTreeKill(this.subProcess.pid, 'SIGTERM').pipe(
+      tap(err => {
+        if (!err) {
```

Comment:
We should handle the case where there is an error.
The biggest risk about this module is that we get a process thats holding onto a port and we're not able to kill it properly.